phpBB/install/app.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+/**
+ * @ignore
+ */
+define('IN_PHPBB', true);
+define('IN_INSTALL', true);
+define('PHPBB_ENVIRONMENT', 'production');
+$phpbb_root_path = '../';
+$phpEx = substr(strrchr(__FILE__, '.'), 1);
+
+//
+// Let's do the common.php logic
+//
+require($phpbb_root_path . 'includes/startup.' . $phpEx);
+require($phpbb_root_path . 'phpbb/class_loader.' . $phpEx);
+
+$phpbb_class_loader = new \phpbb\class_loader('phpbb\\install\\', "{$phpbb_root_path}install/", $phpEx);
+$phpbb_class_loader->register();
+
+$phpbb_class_loader = new \phpbb\class_loader('phpbb\\', "{$phpbb_root_path}phpbb/", $phpEx);
+$phpbb_class_loader->register();
+
+// In case $phpbb_adm_relative_path is not set (in case of an update), use the default.
+$phpbb_adm_relative_path = (isset($phpbb_adm_relative_path)) ? $phpbb_adm_relative_path : 'adm/';
+$phpbb_admin_path = (defined('PHPBB_ADMIN_PATH')) ? PHPBB_ADMIN_PATH : $phpbb_root_path . $phpbb_adm_relative_path;
+
+// Include files
+require($phpbb_root_path . 'includes/functions.' . $phpEx);
+require($phpbb_root_path . 'includes/functions_content.' . $phpEx);
+include($phpbb_root_path . 'includes/functions_compatibility.' . $phpEx);
+require($phpbb_root_path . 'includes/functions_user.' . $phpEx);
+require($phpbb_root_path . 'includes/utf/utf_tools.' . $phpEx);
+
+// Set PHP error handler to ours
+set_error_handler(defined('PHPBB_MSG_HANDLER') ? PHPBB_MSG_HANDLER : 'msg_handler');
+
+$phpbb_installer_container_builder = new \phpbb\di\container_builder($phpbb_root_path, $phpEx);
+$phpbb_installer_container = $phpbb_installer_container_builder
+	->with_environment('installer')
+	->without_extensions()
+	->without_cache()
```

Comment:
It should fail silently but it needs to be tested.

Le sam. 2 mai 2015 23:18, Máté Bartus <notifications@github.com> a écrit :

> In phpBB/install/app.php
> <https://github.com/phpbb/phpbb/pull/3514#discussion_r29550824>:
>
> > +
> > +// Include files
> > +require($phpbb_root_path . 'includes/functions.' . $phpEx);
> > +require($phpbb_root_path . 'includes/functions_content.' . $phpEx);
> > +include($phpbb_root_path . 'includes/functions_compatibility.' . $phpEx);
> > +require($phpbb_root_path . 'includes/functions_user.' . $phpEx);
> > +require($phpbb_root_path . 'includes/utf/utf_tools.' . $phpEx);
> > +
> > +// Set PHP error handler to ours
> > +set_error_handler(defined('PHPBB_MSG_HANDLER') ? PHPBB_MSG_HANDLER : 'msg_handler');
> > +
> > +$phpbb_installer_container_builder = new \phpbb\di\container_builder($phpbb_root_path, $phpEx);
> > +$phpbb_installer_container = $phpbb_installer_container_builder
> > +	->with_environment('installer')
> > +	->without_extensions()
> > +	->without_cache()
>
> Yep, it would be nice, the problem is, at this point we don't know if
> cache is writable. I'm not sure what happens when cache is not writable and
> we try to write it, I'll look into that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phpbb/phpbb/pull/3514/files#r29550824>.
>